src/server/index.ts

```diff
@@ -18,38 +18,49 @@ function analyzer(opts: AnalyzerPluginOptions = { analyzerMode: 'server' }): Plu
   const analyzerModule = createAnalyzerModule(opts?.gzipOptions)
   let defaultWd = process.cwd()
 
+  let previousSourcemapOption: any = false
   const plugin = <Plugin>{
     name,
     apply: 'build',
     enforce: 'post',
     configResolved(config) {
       defaultWd = config.build.outDir ?? config.root
+      previousSourcemapOption = typeof config.build.sourcemap === 'boolean'
+        ? config.build.sourcemap
+        : config.build.sourcemap === 'hidden' ? true : false
+      // https://vitejs.dev/config/build-options.html#build-sourcemap
+      // ensure `sourcemap` option
+      if (!config.build.sourcemap) config.build.sourcemap = 'hidden'
```

Comment:
Seem like incorrect place for fix config value
Replace to `config` hook instead ([link](https://vitejs.dev/guide/api-plugin.html#config))

Reply:
The main problem that I see here: the mutation of a resolved object does have not a clear
Vite provides a resolved config object to every plugin hook without cloned
Mutations in `resolvedConfig` hook a very specific side effects that do not have any predictable results
if I was wrong - sorry. I'd be grateful if you could explain why